File: CodingPractice/src/main/java/Arrays/array/selectionSort.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package array;

/**
 *
 * @author Hp
 */
public class selectionSort 
{
    public static void main(String arg[])
    {
        int arr[] = {3,1, 6, 5};
        
        for(int i = 0; i< arr.length - 1; i++)
        {
            int min = i;
            
            for(int j = i+1; j < arr.length ; j++)
            {
                if(arr[i] > arr[j])
                {
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
            }
        }
        
        for(int i =0 ; i< arr.length ;i++)
        {
            System.out.print(arr[i] + " ");
        }
    }
    
}

File: Python/CodingPractice/LeetCode/Sorting/selectionSort.py
class selectionSort(object):
    def main(self):
        print(self.selectionSort([7,4,6,2,9]))

    def selectionSort(self, arr):
        n = len(arr)

        for i in range(n):

            min_index = i
            for j in range(i+1, n):

                if arr[min_index] > arr[j]:
                    min_index = j

            
            arr[i], arr[min_index] = arr[min_index],arr[i]

        return arr